pkg/daemon/daemon.go

```diff
@@ -254,14 +263,41 @@ func cmdRun(p *ptpProcess) {
 		return
 	}
 
+	if stdoutToSocket {
+		connected = false
+		for {
+			c, err = handleConnection()
+			if err != nil {
+				glog.Errorf("error trying to connect to event socket. retrying in %d secs", connectionRetryInterval)
+				time.Sleep(connectionRetryInterval)
+				continue
+			}
+			connected = true
+			break
+		}
+	}
+
 	done := make(chan struct{})
 
 	scanner := bufio.NewScanner(cmdReader)
 	go func() {
 		for scanner.Scan() {
 			output := scanner.Text()
-			fmt.Printf("%s\n", output)
-			extractMetrics(p.name, p.iface, output)
+			if stdoutToSocket {
+				out := fmt.Sprintf("%s\n", output)
+				fmt.Printf("%s", out)
+				if connected {
+					_, err := c.Write([]byte(out))
+					if err != nil {
+						connected = false
+						glog.Errorf("Write error:", err)
+						go handleConnection()
```

Comment:
Can you explain me here how this will work?
will you be able to continue using the c.Write after the second go routine will be connected?